### DISCUSSION TITLE
Closest Raycast hit with triangle meshes problem

### DISCUSSION THREAD

**Discussion by ArneDJ** [jrouwe/JoltPhysics#440] [Q&A] [answered]
Greetings,

I'm having a small issue when casting rays on a static triangle mesh meant to find the height of a terrain.
The rays are cast above the triangle mesh with the direction downwards, however no matter where the ray is cast from the result is always the same fraction and the same closest hit y-position. 
The hit y-position seems to be the same value as the maximum y-position of the triangle mesh vertices, which leads me to think the ray returns the intersection of the triangle mesh bounding box instead of the triangles within.

Now the thing is that the dynamic objects are colliding with the triangle mesh normally so I don't think the issue is with the triangle mesh itself but rather with the broadphase raycast instead.

To replicate my issue:

Triangle mesh creation:
```C++
auto& body_interface = physics_system.GetBodyInterface();

JPH::BodyCreationSettings settings(new JPH::MeshShapeSettings(vertex_list, index_list), JPH::RVec3(0.f, 0.f, 0.f), JPH::Quat::sIdentity(), JPH::EMotionType::Static, Layers::NON_MOVING);
auto body = body_interface.CreateBody(settings);
body_interface.AddBody(body->GetID(), JPH::EActivation::DontActivate);

triangle_mesh_ids.push_back(body->GetID());

```
Raycast:
```C++
JPH::RayCast ray(from, to - from);

JPH::ClosestHitCollisionCollector<JPH::RayCastBodyCollector> collector;
physics_system.GetBroadPhaseQuery().CastRay(ray, collector, JPH::SpecifiedBroadPhaseLayerFilter(BroadPhaseLayers::NON_MOVING), JPH::SpecifiedObjectLayerFilter(Layers::NON_MOVING));

if (collector.HadHit()) {
	JPH::Vec3 point = ray.GetPointOnRay(collector.mHit.mFraction);
	Console::print("fraction {}", collector.mHit.mFraction);
	Console::print("position x {} y {} z {}", point.GetX(), point.GetY(), point.GetZ());
}
```

**Comment by jrouwe** [comment, marked as answer]
Hello, the issue is that you're doing ```physics_system.GetBroadPhaseQuery()```, this means you're only testing your ray against the broad phase (i.e. the bounding box of the mesh) so it is logical that you always find the same Y position. If you use ```physics_system.GetNarrowPhaseQuery()``` you will go through the broad phase and also the narrow phase and you should hit individual triangles.

**Reply by ArneDJ** [reply]
That fixed my issue, thanks!
